src/runtime_src/core/tools/xbtracer/src/lib/capture.cpp

```diff
@@ -0,0 +1,617 @@
+  {"xrt::run::update_arg_at_index(int, xrt::bo const&)", (void **) &dtbl.run.update_arg2},
+
+  /* kernel class maps */
+  {"xrt::kernel::kernel(xrt::device const&, xrt::uuid const&, std::string const&, xrt::kernel::cu_access_mode)", (void **) &dtbl.kernel.ctor},
+  {"xrt::kernel::kernel(xrt::hw_context const&, std::string const&)",  (void **) &dtbl.kernel.ctor2},
+  {"xrt::kernel::group_id(int) const",  (void **) &dtbl.kernel.group_id},
+  {"xrt::kernel::offset(int) const",  (void **) &dtbl.kernel.offset},
+  {"xrt::kernel::write_register(unsigned int, unsigned int)",  (void **) &dtbl.kernel.write_register},
+  {"xrt::kernel::read_register(unsigned int) const",  (void **) &dtbl.kernel.read_register},
+  {"xrt::kernel::get_name() const",  (void **) &dtbl.kernel.get_name},
+  {"xrt::kernel::get_xclbin() const",  (void **) &dtbl.kernel.get_xclbin},
+
+  /* xclbin class maps */
+  {"xrt::xclbin::xclbin(std::string const&)", (void **) &dtbl.xclbin.ctor_fnm },
+  {"xrt::xclbin::xclbin(axlf const*)", (void **) &dtbl.xclbin.ctor_axlf },
+  {"xrt::xclbin::xclbin(std::vector<char, std::allocator<char> > const&)", (void **) &dtbl.xclbin.ctor_raw},
+
+  /*hw_context class maps*/
+  {"xrt::hw_context::hw_context(xrt::device const&, xrt::uuid const&, xrt::hw_context::cfg_param_type const&)", (void **) &dtbl.hw_context.ctor_frm_cfg},
+  {"xrt::hw_context::hw_context(xrt::device const&, xrt::uuid const&, xrt::hw_context::access_mode)", (void **) &dtbl.hw_context.ctor_frm_mode},
+  {"xrt::hw_context::update_qos(xrt::hw_context::cfg_param_type const&)", (void **) &dtbl.hw_context.update_qos},
+  // NOLINTEND(cppcoreguidelines-pro-type-cstyle-cast)
+};
+
+} //namespace xrt::tools::xbtracer
+
+#ifdef __linux__
+#include <cxxabi.h>
+#include <dlfcn.h>
+#include <elf.h>
+#include <unistd.h>
+
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+
+namespace xrt::tools::xbtracer {
+
+constexpr const char* lib_name = "libxrt_coreutil.so";
+
+// mutex to serialize dlerror and getenv
+// NOLINTBEGIN(cppcoreguidelines-avoid-non-const-global-variables)
+std::mutex dlerror_mutex;
+extern std::mutex env_mutex;
+// NOLINTEND(cppcoreguidelines-avoid-non-const-global-variables)
+
+/**
+ * This class will perform following operations.
+ * 1. Read mangled symbols from .so file.
+ * 2. perform demangling operation.
+ * 3. update xrt_dtable which will be used to invoke original API's
+ */
+class router
+{
+  private:
+  void* handle = nullptr;
+  /* library Path */
+  std::string path;
+  /*
+   * This will create association between function name (key)
+   * and mangled function name (value).
+   */
+  std::unordered_map<std::string, std::string> func_mangled;
+
+  public:
+  router(const router&);
+  auto operator = (const router&) -> router&;
+  router(router && other) noexcept;
+  auto operator = (router&& other) noexcept -> router&;
+  auto load_func_addr() -> int;
+  auto load_symbols() -> int;
```

Comment:
In general `-> return_type` doesn't read as well as listing the return_type in place of auto.   There are only very few cases where trailing return_type is needed and when not, the syntax is not wide adopted in C++.

Reply:
Was handled in 9/25 update